### PR TITLE
Add docs to ConnManager interface

### PR DESCRIPTION
**Pull request by raulk**
Noticed that this interface was missing docs, so I took the liberty to add them. Let me know if something appears to be inaccurate!

### CODE REVIEW

**Comment by raulk** [comment]
@whyrusleeping adding you as a reviewer given that you're the original author!

File: interface.go [hunk 1]
```diff
@@ -5,20 +5,42 @@ import (
...
 )
 
+// ConnManager tracks connections to peers, and allows consumers to associate metadata
+// with each peer.
+//
+// It enables connections to be trimmed based on implementation-defined heuristics.
 type ConnManager interface {
+
+	// TagPeer sets a tag value for the given peer.
+	// A tag can be a distance, latency, reputation, etc. value.
```
Comment (Stebalien) [inline interface.go]: Tag peer really:

1. Tags the peer with the string.
2. Associates a weight with that tag.

latency, distance, etc are probably not good weights.

Comment (raulk) [reply interface.go]: 👍 Applied the changes.

**Comment by raulk** [comment]
@Stebalien – I think we're ready here!

**Comment by Stebalien** [comment]
Thanks!

**Pull request merged**